src/main/java/training/bookish/configuration/SecurityConfig.java

```diff
@@ -0,0 +1,49 @@
+package training.bookish.configuration;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+
+import javax.sql.DataSource;
+
+@Configuration
+public class SecurityConfig extends WebSecurityConfigurerAdapter {
+
+    @Autowired

```

Comment:
It's worth knowing that there are other ways to configure dependency injection in Spring. You can add `@Autowired` to a field like you've done here, or you can add it to the constructor:

```
private final Datasource datasource;

@Autowired
public SecurityConfig(Datasource datasource) {
  this.datasource = datasource;
}
```

This does exactly the same thing, but it has two advantages:

* You can make the field final
* You can call the constructor from a test, which (I think) makes the test clearer than configuring Spring dependency injection in the test

This might make it easier to set up the integration tests later.